tests/__init__.py

```diff
@@ -109,6 +109,9 @@ def from_mm(dpi, x=0, y=0):
 
 
 class TestBase(object):
+    def teardown(self):
+        sleep(3)
```

Comment:
whoah. this needs to wait for the real thing we're waiting for. udev remove events?

Reply:
See https://gitlab.freedesktop.org/libevdev/hid-tools/issues/19

This delay allows `hidtools` to catch up, making the tests more reliable, at least on the machines I tested.

Reply:
don't we need to call `dispatch()` to make it catch up? or is that running in the other thread anyway.

Either way, this is where you add massive comment explaining *why* this is needed, random sleeps somewhere are hard to understand otherwise. Also - 3 seconds? That's a crazy long timeout.

Reply:
That is running in a different process altogether. Here we are in the test suite, we are just a client. The device handling, including calling `dispatch()`, is done by the server.

I think the main issue is that we are calling dispatch from just 1 thread. Benjamin says `hid-tools` is not designed for multi-threading and it would require a major refactoring.

You should see the discussion bellow, Benjamin said this should be removed.